src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(value) {
+    if (value || value === 0) {
+      sum += value;
+
+      return adder;
+    } else {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+  }
```

Comment:
The `adder` function should not check for the presence of a value using `value || value === 0`. According to the task description, there is no need to check that the argument passed to the `adder` function is a number, and there is no need to set a default value for the `adder` function argument. Instead, you should check if the function is called without any arguments to return the result and reset the sum.